src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -75,16 +75,28 @@ default FilePath getFilePathForUniqueFolder(Launcher launcher, String uniqueName
         /*Use of Computer is not recommended as jenkins hygeine for pipeline support
          * https://javadoc.jenkins-ci.org/jenkins/tasks/SimpleBuildStep.html */
         
-        String tmpDir = getNodeSpecificTmpFolderPath(workspace);
-        return new FilePath(launcher.getChannel(), tmpDir+"/"+uniqueName);
+        String tmpDir = getNodeSpecificTmpFolderPath(workspace, launcher);
+        String fileSeperator = "/";
+
+        if (!launcher.isUnix()){
+            fileSeperator = "\\";
+        }
+
+        return new FilePath(launcher.getChannel(), tmpDir + fileSeperator + uniqueName);
     }
 
-    default String getNodeSpecificTmpFolderPath(FilePath workspace) throws IOException, InterruptedException {
+    default String getNodeSpecificTmpFolderPath(FilePath workspace, Launcher launcher) throws IOException, InterruptedException {
         Computer cmp = workspace.toComputer();
         if (cmp == null) {
             throw new IOException(Message.getValue("build.workspace.computer.not.found"));
         }
         String tmpDir = (String) cmp.getSystemProperties().get("java.io.tmpdir");
+
+        // Handle Java temp folder path discrepancy for Windows.
+        if (!launcher.isUnix() && tmpDir.charAt(tmpDir.length() - 1) == '\\'){
+            tmpDir = tmpDir.substring(0, tmpDir.length() - 1);
+        }

```

Comment:
Why only handle this case for Windows? I believe `java.io.tmpdir` is configurable and may have a file separator at the end on any system.

Reply:
I think that the difference here java.io.tmpdir return `/tmp`  on Linux and `C:\Users\nbhoski\AppData\Local\Temp\`  on Windows, It adds  the trailing backslash on windows by default but not on Linux.

Reply:
But as far as I know there is no reason java.io.tmpdir could not return `/tmp/` on Linux. java.io.tmpdir is user configurable.

Reply:
I updated the code to avoid trimming string and use FilePath.getRemote() instead. FilePath object invokes 'normalize' function which handles extra file separators cross platform.